rclcpp/src/rclcpp/executor.cpp

```diff
@@ -108,20 +108,37 @@ Executor::spin_node_some(rclcpp::node::Node::SharedPtr node)
 void
 Executor::spin_some()
 {
-  while (AnyExecutable::SharedPtr any_exec =
-    get_next_executable(std::chrono::milliseconds::zero()))
-  {
+  if (spinning.exchange(true)) {
+    throw std::runtime_error("spin_some() called while already spinning");
+  }
+  AnyExecutable::SharedPtr any_exec;
+  while ((any_exec = get_next_executable(std::chrono::milliseconds::zero())) && spinning.load()) {
     execute_any_executable(any_exec);
   }
+  spinning.store(false);
```

Comment:
Sorry, I just realized you were expecting me to do this :smile:. I totally misread it, but anyways I just pushed my fix. I did it by implementing what's called a scope exit or scope guard. Boost has a scope exit and so does the D language. There's some interesting discussion about including something like this in the c++ std library at some point. Anyways this should work us and might be useful else where too.